src/Compilers/CSharp/Test/Symbol/Symbols/ModuleInitializersTests.Targets.cs

```diff
@@ -0,0 +1,508 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Symbols
+{
+    public sealed partial class ModuleInitializersTests
+    {
+        [Fact]
+        public void TargetMustNotBeLocalFunction()
+        {
+            string source = @"
+using System.Runtime.CompilerServices;
+
+class C
+{
+    internal static void M()
+    {
+        LocalFunction();
+        [ModuleInitializer]
+        static void LocalFunction() { }
+    }
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+            var compilation = CreateCompilation(source, parseOptions: s_parseOptions);
+            compilation.VerifyEmitDiagnostics(
```

Comment:
Was there a reason we call 'VerifyEmitDiagnostics' throughout here? If we want to make sure the diagnostics will pop up interactively (i.e. without having to hit build in VS) we usually call 'VerifyDiagnostics'.

Reply:
> Was there a reason we call 'VerifyEmitDiagnostics' throughout here? 

I explicitly suggested doing this, you can find that in the PR comments.

> If we want to make sure the diagnostics will pop up interactively (i.e. without having to hit build in VS) we usually call 'VerifyDiagnostics'.

It is a known fact that declaration diagnostics pop ups interactively 

---
In reply to: [422288481](https://github.com/dotnet/roslyn/pull/43301#discussion_r422288481) [](ancestors = 422288481)

Reply:
Both things are nice to be aware of, thanks!